### PR TITLE
New db schema.

### PR DESCRIPTION
**Pull request by JeldorPKU**
Add new columns and fixed the token bug.

### CODE REVIEW

File: capurace.sql [hunk 1]
```diff
@@ -77,7 +84,15 @@ CREATE TABLE IF NOT EXISTS `users` (
...
   `activated` tinyint(1) NOT NULL DEFAULT '0',
   `token` varchar(32) NOT NULL
-) ENGINE=InnoDB AUTO_INCREMENT=21 DEFAULT CHARSET=utf8;
+) ENGINE=InnoDB AUTO_INCREMENT=27 DEFAULT CHARSET=utf8;
+
+--
+-- Dumping data for table `users`
+--
+
+INSERT INTO `users` (`id`, `school`, `leader`, `tel`, `mail`, `password`, `bill`, `paid`, `confirmed`, `association_name`, `province`, `address`, `zipcode`, `activated`, `token`) VALUES
```
Comment (huxuan) [inline capurace.sql]: Please remove the data before dump.

File: application/controllers/user.php [hunk 2]
```diff
@@ -89,8 +89,10 @@ public function signup() {
             } else {
                 $err_code = '200';
                 unset($data['passconf']);
+                $token = $this->user->set_token($data['mail']);
```
Comment (huxuan) [inline application/controllers/user.php]: It's great to set token in this order, but I prefer we replace `set_token()` to something like `get_token()` and it does nothing to the database. Currently, `set_token()` try to update the database and that not what we want before inserting the user.

**Comment by huxuan** [comment]
Seems this PR contains many other changes. I would suggest you to split it. ;-)

**Comment by JeldorPKU** [comment]
Those are the commits from the previous PR. Shall I split the token bug?

**Comment by huxuan** [comment]
I think `git fetch upstream && git rebase upstream/master && git push -f` should do the work.
It's ok to contain the fixing of token bug, only if you add that in the PR title or description. ;-)

**Pull request merged**